packages/webdriverio/src/commands/element/addValue.js

```diff
@@ -32,5 +32,5 @@ export default function addValue (value) {
         return this.elementSendKeys(this.elementId, transformToCharString(value))
     }
 
-    return this.elementSendKeys(this.elementId, value)
+    return this.elementSendKeys(this.elementId, value, transformToCharString(value))
```

Comment:
how come elementSendKeys take three arguments?

Reply:
@nami-varthakavi I've added one more parameter to "/session/:sessionId/element/:elementId/value" in https://github.com/webdriverio/webdriverio/pull/3428/files#diff-e4d79ef8dfe79517f19a6960718bfa10

Reply:
I would say might as well have another command with ```{
        "name": "value",
        "type": "string[]",
        "description": "The sequence of keys to type. An array must be provided.",
        "required": true
      }]	      }]```

Reply:
We can choose between jsonwp and webdriver command in this case.
Jsonwp scenario is unchanged, I've only changed webdriver's command to isolate workaround as much as possible.
Change doesn't affect w3c usage of value command but adds compatibility with jsonwp value command

Reply:
it is not possible to have two commands for same endpoint and I don't want to add such logic for temporary workaround. Also, I don't want to override W3C command with JSONWP command even for Appium, however if you insist, I can change ProtocolCommands setup and override W3C "/session/:sessionId/element/:elementId/value" with JSONWP "/session/:sessionId/element/:elementId/value".

Reply:
I understand this is a workaround but I would probably not change the basic structure of existing command. Rather go by the above approach i mentioned and make it is only applicable to JSONWP. we don't complicate things and understand clearly what is the purpose of both commands.

Reply:
I'm not changing command, I'm extending it.

Reply:
Appium uses W3C, any changes to JSONWP won't make any effect

Reply:
I might be ok with `however if you insist, I can change ProtocolCommands setup and override W3C "/session/:sessionId/element/:elementId/value" with JSONWP "/session/:sessionId/element/:elementId/value".` but I would @christian-bromann should give a final say on this.

Reply:
my first PR was overriding W3C with JSONWP commands for Appium.

Reply:
https://github.com/webdriverio/webdriverio/pull/3425

Reply:
> I might be ok with `however if you insist, I can change ProtocolCommands setup and override W3C "/session/:sessionId/element/:elementId/value" with JSONWP "/session/:sessionId/element/:elementId/value".` but I would say @christian-bromann should give a final say on this.

It would be really ugly solution. I had to change ProtocolCommands logic and add condition to addValue:
```
export default function addValue (value) {
    if (!this.isW3C || this.isMobile) { // added  || this.isMobile
        return this.elementSendKeys(this.elementId, transformToCharString(value))
    }

    return this.elementSendKeys(this.elementId, value, transformToCharString(value))
}
```

Reply:
I added this solution before. It was reverted back :(

Reply:
https://github.com/webdriverio/webdriverio/pull/3347

Reply:
@nami-varthakavi your solution causing failure for me:
```
Error: Malformed type for "text" parameter of command elementSendKeys
Expected: string
Actual: object
```

Reply:
which is absolutely obvious because **parameter type check** validates parameters against W3C command that accept string, and you are sending an array.

Reply:
I'd better say that architecture is quite solid so that we can apply hacky workarounds easily, which is good.